### PR TITLE
HAR-7987 - track changes

### PR DESCRIPTION
**Pull request by artem-harbour**
**BEFORE MERGE:**
- Let me know what to do with the test buttons in the toolbar.
- I think we should discuss document normalization first.

List of changes:
- Track changes has been completely redesigned from scratch.
`packages/super-editor/src/extensions/track-changes/`
- The previous implementation has been moved to the `track-changes-deprecated` path for reference only and will be removed over time.
`packages/super-editor/src/extensions/track-changes-deprecated/`
- Fixed implementation of track changes marks.
- Fixed commands for accepting and rejecting tracked changes by selection (Also can be used ID).
- Fixed ID assignment when adding or deleting text. Using `findTrackedMarkBetween` function.
`packages/super-editor/src/extensions/track-changes/trackChangesHelpers/findTrackedMarkBetween.js`
- Changed approach with `amendTransaction` function and removal of tracked changes if plugin is disabled. This will lead to various complications and makes no sense. Instead we use the `trackedTransaction` function if the plugin is enabled or simply return `tr` (transaction) without any changes.
- Implemented track changes for formatting.
- Other issues have been fixed, such as when we remove tracked text inside an added.

Notes:
- Track changes need to be connected to comments.
- There is a need to discuss the issue of document normalization so that it works well for users' documents.
- Currently Chrome has some issue in the latest update that track changes does not always work as expected, in other browsers everything is fine (**looks like I found a workaround for this**).

### CODE REVIEW

File: packages/super-editor/src/core/helpers/createDocument.js [hunk 1]
```diff
@@ -6,8 +8,18 @@
  */
 export function createDocument(converter, schema) {
   const documentData = converter.getSchema();
-  if (documentData) {
-    return schema.nodeFromJSON(documentData);
+
+  // Experimental.
+  let normalizedData;
+  try {
+    normalizedData = normalizeDocument(documentData);
```
Comment (artem-harbour) [inline packages/super-editor/src/core/helpers/createDocument.js:15]: This is an item for discussion before the merge.

File: packages/super-editor/src/core/Editor.js [hunk 2]
```diff
@@ -519,13 +520,23 @@ export class Editor extends EventEmitter {
 
     let state;
     try {
-      const trackedTr = amendTransaction(transaction, this.view, this.options.user);
-      const { state: newState } = this.view.state.applyTransaction(trackedTr);
+      const trackChangesState = TrackChangesBasePluginKey.getState(this.view.state);
+      const isTrackChangesActive = trackChangesState?.isTrackChangesActive ?? false;
+
+      const tr = isTrackChangesActive
```
Comment (harbournick) [inline packages/super-editor/src/core/Editor.js:526]: much better!

File: packages/super-editor/src/core/helpers/createDocument.js (hunk 1 above)
Comment (harbournick) [reply packages/super-editor/src/core/helpers/createDocument.js:15]: leaving it in for now, until import refactor

File: packages/super-editor/src/extensions/track-changes/trackChangesHelpers/documentHelpers.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+
+  if (toArr) {
+    return marksFound;
+  }
+
+  return markFound;
+};
+
+// https://discuss.prosemirror.net/t/expanding-the-selection-to-the-active-mark/
+export const findMarkPosition = (doc, pos, markName) => {
```
Comment (harbournick) [inline packages/super-editor/src/extensions/track-changes/trackChangesHelpers/documentHelpers.js:34]: ah, fascinating

**Pull request merged**